pkg/client/unversioned/restclient.go

```diff
@@ -45,26 +44,24 @@ const (
 //
 // Most consumers should use client.New() to get a Kubernetes API client.
 type RESTClient struct {
-	baseURL *url.URL
-	// A string identifying the version of the API this client is expected to use.
-	groupVersion unversioned.GroupVersion
+	// Base is the root URL for all invocations of the client
+	Base *url.URL
```

Comment:
Really just to simplify down the struct.  Private is great when we expect to have all private - likewise public is great when we don't expect the object to change.  I like putting a bit of pressure on us to treat RESTClient implementation as a public thing (like http.Client) vs. an interface (like RoundTripper), at least in terms of how we instantiate it.